node/src/components/linear_chain.rs

```diff
@@ -193,20 +241,24 @@ where
                 },
             },
             Event::LinearChainBlock {
-                block,
+                mut block,
                 execution_results,
-            } => effect_builder
-                .put_block_to_storage(block.clone())
-                .event(move |_| Event::PutBlockResult {
-                    block,
-                    execution_results,
-                }),
+            } => {
+                let mut effects = self.add_pending_finality_signatures(&mut block, effect_builder);
+                effects.extend(effect_builder.put_block_to_storage(block.clone()).event(
+                    move |_| Event::PutBlockResult {
+                        block,
+                        execution_results,
+                    },
+                ));
+                effects
+            }
             Event::PutBlockResult {
                 block,
                 execution_results,
             } => {
                 // TODO: Remove once we can return all linear chain blocks from persistent storage.
-                self.linear_chain.push(*block.clone());
+                self.latest_block = Some(*block.clone());
```

Comment:
I think the `TODO` above can be removed as well.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/649/commits/8404eced79d3716e3393a238fa2d18ad6a60db17.